### PR TITLE
Add support of MacOS

### PR DESCRIPTION
**Pull request by essobedo**
## Motivation

To be able to execute tests on GitHub runners that require native compilation, we have to be able to launch them against [a MacOS runner because it has 14 Go of RAM while only 7 Go is available on Linux runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) which is not enough for the new tests that need up to 9.5 Go of RAM.

## Modifications:

* Install docker using `brew` since the version of docker is not part of the input parameters of the action
* Download the binaries of `kubectl` and `kind` corresponding to the local arch
* Check the size of the list before calling them to avoid getting `unbound variable` error
* Avoid using `tee` to append a line to a file since the command is different between MacOS and Linux
* Add `macos-latest` to the OSs to be tested by the CI
* Print the version of `kubectl` and `kind` once installed to ensure that the right version has been installed

### CODE REVIEW

**Comment by essobedo** [comment]
@lburgazzoli @johnpoth Could you please review it? 🙏 Thanks in advance

**Review by oscerd — APPROVED** [review]

**Comment by essobedo** [comment]
@oscerd Hi, thx for approving, do you have enough rights to merge it, or maybe you prefer to wait for another review?

File: kind.sh
Comment (lburgazzoli) [inline kind.sh]: I think this change the behavior of the action as it would install whatever kind version is defined in brew regardless of the kind version that is set in the action.

File: kind.sh
Comment (lburgazzoli) [inline kind.sh]: I think this change the behavior of the action as it would install whatever kubectl version is defined in brew regardless of the kubectl version that is set in the action.

File: kind.sh
Comment (essobedo) [reply kind.sh]: Good point, let me change that 👍

File: kind.sh
Comment (lburgazzoli) [inline kind.sh]: there should be a space I think between ] and &&

**Review by lburgazzoli — APPROVED** [review]

**Review by squakez — APPROVED** [review]

**Comment by essobedo** [comment]
Should be good now,[ I have to test it first](https://github.com/essobedo/test-action/actions/runs/4122844694) to ensure that it works as expected which allowed me to find some bugs. If it is good for you, could you please merge it 🙏 .

Thx all for your reviews and comments.

**Comment by essobedo** [comment]
I see that some tests are failing, I will propose another PR to fix them

**Pull request merged**
